homeassistant/components/touchline/config_flow.py

```diff
@@ -0,0 +1,104 @@
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.data_entry_flow import FlowResult
+import homeassistant.helpers.config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+RESULT_SUCCESS = "success"
+RESULT_CANNOT_CONNECT = "cannot_connect"
+
+
+def _try_connect_and_fetch_basic_info(host):
+    """Attempt to connect and, if successful, fetch number of devices."""
+    py_touchline = PyTouchline()
+    result = {"type": None, "data": {}}
+    number_of_devices = None
+    try:
+        number_of_devices = int(py_touchline.get_number_of_devices(host))
+        if number_of_devices:
+            result["type"] = RESULT_SUCCESS
+            result["data"] = None
+
+            _LOGGER.debug(
+                "Number of devices found: %s",
+                number_of_devices,
+            )
+            device = PyTouchline(id=0)
+            device.update()
+            controller_id = device.get_controller_id()
+            result["data"] = controller_id
+            return result
+    except ConnectionRefusedError:
+        _LOGGER.debug(
+            "Failed to connect to device %s. Check the IP address "
+            "as well as whether the device is connected to power and network",
+            host,
+        )
+        result["type"] = RESULT_CANNOT_CONNECT
+    return result
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Roth Touchline."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        if user_input is None:
+            return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+        # Abort if an entry with same host and port is present.
+        self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+
+        errors = {}
+        host = user_input[CONF_HOST]
+        # Remove HTTPS and HTTP from URL
+        pattern = "https?://"
+        user_input[CONF_HOST] = re.sub(pattern, "", user_input[CONF_HOST])
+        user_input[CONF_HOST] = "http://" + user_input[CONF_HOST]
+        if not cv.url(user_input[CONF_HOST]):
+            errors["base"] = "invalid_input"
+        else:
+            self._async_abort_entries_match({CONF_HOST: host})
+            result = await self.hass.async_add_executor_job(
+                _try_connect_and_fetch_basic_info, user_input[CONF_HOST]
+            )
+
+            if result["type"] != RESULT_SUCCESS:
+                errors["base"] = "cannot_connect"
+
+        if errors:
+            return self.async_show_form(
+                step_id="user", data_schema=DATA_SCHEMA, errors=errors
+            )
+
+        await self.async_set_unique_id(result["data"])
```

Comment:
What is an example value of `result["data"]`?

Reply:
The values I have seen are 98 and 224, they are part of a number in the controller. I have two controllers, one has 98 and the other 224 as "controller ID". I doubt that they are really unique (should have been a much longer number then), but to the end user, I believe they are.
If this is not good enough, I guess the only option is to drop creating a unique id?

Reply:
Your call—I'm not familiar with this device. As you point out, a unique ID shouldn't change; if that's the case here, you're good!